### PR TITLE
feat: implement artifacts preview for store

### PR DESCRIPTION
**Pull request by adong**
## Context

Making necessary changes to implement artifacts for store.
Instead of receiving `download=true`, now query params will be receiving  `type=preview` or `type=download`.

If query paramter is `type=preview`, then response header's MIME type will be `text/html` for html files.

## Objective

Store is the gateway for fetching artifacts files, in order to implement preview feature in the UI, we need the store to display correct mime types.

## References

UI: https://github.com/screwdriver-cd/ui/pull/465
Backend: https://github.com/screwdriver-cd/screwdriver/pull/1703
Store: https://github.com/screwdriver-cd/store/pull/88

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/builds.js [hunk 1]
```diff
@@ -87,11 +118,32 @@ exports.plugin = {
                     }
                 }
 
+                const fileName = artifact.split('/').pop();
+                const fileExt = fileName.split('.').pop();
+                let mime = getMimeFromFileExtension(fileExt);
+
+                if (value && !value.h && value.type === 'Buffer') {
```
Comment (minz1027) [inline plugins/builds.js]: guess we don't need it here since the only type we care about is html

Comment (adong) [reply plugins/builds.js]: Removed

File: helpers/iframe.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        urlFileDir = urlFileDir.slice(2, urlFileDir.length-1).join('/');
+
+        const apiUrl = new URL(apiHref);
+        const apiUrlOrigin = apiUrl.origin;
+        const apiVersion = apiUrl.pathname.split('/')[1] || '${apiVersion}';
+
+        const anchors = document.getElementsByTagName('a');
+        for (let anchor of anchors) {
+          let originalHref = anchor.attributes.href.value;
+          const newUrl = apiUrlOrigin + '/' + apiVersion + '/' + urlFileDir + '/' + originalHref;
```
Comment (tkyi) [inline helpers/iframe.js]: encodeURI?

File: helpers/mime.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+/**
+ * getMimeFromFileExtension
+ * @param  {String} file extension, like css, txt, html
```
Comment (tkyi) [inline helpers/mime.js]:
```suggestion
 * @param  {String} fileExtension  File extension (e.g. css, txt, html)
```

File: helpers/iframe.js (hunk 2 above)
Comment (adong) [reply helpers/iframe.js]: This function does string concatenation to mask the urls to be prefix-ed from api.

File: helpers/mime.js (hunk 3 above)
Comment (adong) [reply helpers/mime.js]: Wow, github now allows directly. 

image

File: plugins/builds.js [hunk 4]
```diff
@@ -87,11 +93,23 @@ exports.plugin = {
                     }
                 }
 
+                const fileName = artifact.split('/').pop();
```
Comment (minz1027) [inline plugins/builds.js]: @tkyi @adong can add the encodeURI here. And change this line to use fileName: https://github.com/screwdriver-cd/store/blob/00fa98be5b377907d32baa2870ddbe25aa9b7a69/plugins/builds.js#L104

File: test/plugins/builds.test.js [hunk 5]
```diff
@@ -186,7 +186,7 @@ describe('builds plugin test', () => {
             assert.equal(getResponse.result, 'THIS IS A TEST');
 
             const downloadResponse = await server.inject({
-                url: `/builds/${mockBuildID}/foo?download=true`,
+                url: `/builds/${mockBuildID}/foo?type=download`,
```
[the remark this answers is not in the document]
Comment (adong) [reply test/plugins/builds.test.js]: This test cases just came from latest rebase. 

I had added `type=preview` test cases in

1) https://github.com/screwdriver-cd/store/pull/88/files#diff-d6ca80dbacb96b4afde6b87e44b043c5R121

2) https://github.com/screwdriver-cd/store/pull/88/files#diff-d6ca80dbacb96b4afde6b87e44b043c5R150

File: plugins/builds.js (hunk 4 above)
Comment (adong) [reply plugins/builds.js]: Sounds good.

**Comment by tk3fftk** [comment]
ref: https://github.com/screwdriver-cd/screwdriver/issues/1501

**Pull request merged**